packages/config/src/ProfileInfo.ts

```diff
@@ -863,11 +865,59 @@ export class ProfileInfo {
      *      into mergedArgs.knownArgs if a value is found in an environment
      *      variable for any missingArgs.
      */
-    private overrideWithEnv(mergedArgs: IProfMergedArg) {
-        if (this.mOverrideWithEnv === false) {
-            return;
-        }
+    private overrideWithEnv(mergedArgs: IProfMergedArg, profSchema?: IProfileSchema) {
+        if (!this.mOverrideWithEnv) return; // Don't do anything
+
+        // Populate any missing options
+        const envPrefix = ImperativeConfig.instance.loadedConfig.envVariablePrefix;
+        const envStart = envPrefix + "_OPT_";
+        for (const key in process.env) {
+            if (key.startsWith(envStart)) {
+                let argValue: any = process.env[key];
+                let dataType: any = typeof argValue;
+                const argName: string = CliUtils.getOptionFormat(key.substring(envStart.length).replace(/_/g, "-").toLowerCase()).camelCase;
+
+                let argNameFound = false;
+                if (profSchema != null) {
+                    for (const [propName, propObj] of Object.entries(profSchema.properties || {})) {
+                        if (argName === propName) {
+                            dataType = this.argDataType(propObj.type);
+                            argNameFound = true;
+                        }
+                    }
+                }
 
-        // todo: get values from environment variables
+                if (profSchema == null || !argNameFound) {
```

Comment:
Is it desired behavior that I can set a property `ZOWE_OPT_SOME_ARG_NAME_THAT_DOES_NOT_EXIST`, and the value will get read as a known arg?

Just want to make sure we don't get bitten by a scenario like this:
```shell
export ZOWE_OPT_RECURSIVE=true
zowe files delete uss <dirname> -f
```

Oh crap, I deleted a whole directory recursively!

I know this is an odd scenario - why would someone set this environment variable? But it could be unexpected behavior if users expect only properties contained in the profile schema (host, user, password, etc) to be loaded from environment variables.

Reply:
This may also be a broader discussion than just environment variables - I suppose users could also set `'recursive': true` in a config JSON file, which would result in a similar scenario.

Reply:
I believe that our agreement with consumers of this API was that we would only process properties that were in a profile (or missing properties that were defined for that profile type). I think that we should extend that to environment variables.

I think would could achieve that limitation if some of this logic were grouped under the following test, which occurs earlier in function:
```
       if (profSchema != null) {
```
Inside there, we could test that the environment variable matches a missing arg.
I think that we would still want to process environment variables for the knownArgs even if we did not have a schema.

Reply:
> I believe that our agreement with consumers of this API was that we would only process properties that were in a profile (or missing properties that were defined for that profile type). I think that we should extend that to environment variables.

I agree with this - but it's unclear to me how we will "only process properties that were in a profile" when loading arg values from a config JSON file? If there is no schema present, a user could define any args they wish in a team config profile (`response-format-json`, `help-web`, etc).

Another question - if there is a schema present, should `mergeArgsForProfile` throw an error when non-profile args (like `rfj`) are defined in a config JSON file? This validation isn't happening currently.

> I think that we would still want to process environment variables for the knownArgs even if we did not have a schema.

I think environment variables only need to be processed for missingArgs. The way knownArgs is currently implemented, all the args it contains have values already defined, so looking at environment variables (at the bottom of the order of precedence list) could be skipped.

Reply:
I've made some changes to the implementation...
Please see comment below

Reply:
I think the placement of default values into either missing or known args muddies the conversation. Currently default values are in missing, but our VSCE users could ask to change that. I think that these statements follow the order-of-precedence used by the CLI, and mostly conforms to Timothy's guidelines. 

1. A default value can be overridden by an env var. We only get default values if we can find a schema.
2. A missing value (which is not default) can also be overridden by an env var.  We only get missing values by finding a schema and finding properties that are required.
3. A known value should NOT be overridden by an env var. Just like the CLI, if the user places a value in a config file, the config file takes precedence over an env var. Sorry, I forgot this particular rule in earlier conversations.

Reply:
hopefully this helps address the 3rd rule 😋 
https://github.com/zowe/imperative/blob/c352d576bc372b551358ca6aee1861674e32b32d/packages/config/__tests__/ProfileInfo.test.ts#L329-L334

Reply:
each property tries to address a different scenario.
- `host`: property from team config -> ENV var will not update it even if `mOverrideWithEnv === true`
- `RFH `: property from team config -> not found in schema, nor missing arguments (Not updated)
- `port`: property defined in schema with default value and not specified in team config
- `list`: property defined in schema but not specified in team config

Reply:
Thanks for addressing and sorry for complicating this discussion by getting the behavior of `mergeArgsForProfile` involved 😢 I'll move that part of the discussion to #579, so this conversation can be resolved 🙂

Reply:
no worries! Thanks for the discussion! : )